run-tests.pl

```diff
@@ -448,21 +448,37 @@ sub fixture
    # actually invoke $setup
    @req_futures or push @req_futures, $f_start;
 
+   my $start = sub {
+      return if $f_start->is_ready;   # already started
+      $OUTPUT->diag( "Starting fixture $name" ) if ( $VERBOSE >= 2 );
+      $f_start->done( @_ );
+   };
+
    return Fixture(
       $name,
 
       \@requires,
 
-      sub { $f_start->done( @_ ) unless $f_start->is_ready },
+      $start,
 
       Future->needs_all( map { without_cancel($_) } @req_futures )
-         ->then( $setup )
-         ->set_label( $name ),
+         ->on_fail( sub {
+            my ( $reason ) = @_;
+
+            warn( "$name: input fixture failed with $reason" )
+               if ( $VERBOSE >= 3 );
```

Comment:
I'm surprised we seem to be warning that things failed at a higher verbosity level than when it successfully finishes.

Reply:
Also, would it be better to have it after `->then( $setup )` in case that fails? Or are we also logging on failure further down?

Reply:
yeah, the reason for both (the fact it's only logged at high verbosity, and that `$setup` failures aren't logged here) is that both situations will make this fixture fail, which will be logged at https://github.com/matrix-org/sytest/pull/397/files/88138b6705dcbf70a484ce8b3d6c5886e6535280#diff-2bd42e0e657c22140d1cc8bf57bbad95R628.